### PR TITLE
Terraform Example Code Optimized

### PR DESCRIPTION
**Pull request by marifse**
### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

**Comment by marifse** [comment]
@ocofaigh can you kindly review this PR and approve. Thanks

**Comment by marifse** [comment]
@ocofaigh can you please rerun

**Review by ocofaigh — COMMENTED** [review]
@marifse One of the goals of this exercise is to remove variables from the examples where possible. I see `examples/default/variables.tf` has other variables that myabe can be moved into the example directly?

File: examples/default/main.tf [hunk 1]
```diff
@@ -21,7 +21,7 @@ module "vpc" {
   resource_group_id = module.resource_group.resource_group_id
   region            = var.region
   prefix            = var.prefix
-  name              = var.vpc_name
+  name              = "${var.prefix}-vpc"
```
Comment (ocofaigh) [inline examples/default/main.tf:24]: set this to `name = "vpc-instance"`

File: examples/default/main.tf [hunk 2]
```diff
@@ -79,7 +79,7 @@ module "vpes" {
   source             = "../../"
   region             = var.region
   prefix             = var.prefix
-  vpc_name           = var.vpc_name
+  vpc_name           = "${var.prefix}-vpc"
```
Comment (ocofaigh) [inline examples/default/main.tf:82]: set this to `name = "vpc-instance"`

**Review by ocofaigh — CHANGES_REQUESTED** [review]
see comments

**Comment by marifse** [comment]
@ocofaigh please check now.

**Comment by ocofaigh** [comment]
@toddgiguere Is this a known issue?
```
TestRunEveryMultiTenantExample 2024-04-03T16:19:39Z logger.go:66: │ Error: "name" ("vpe-allmt-zlrco2-vpe-allmt-zlrco2-vpc-instance-vpc-hs-crypto-tke-2") must contain from 1 to 63 characters 
TestRunEveryMultiTenantExample 2024-04-03T16:19:39Z logger.go:66: │ 
TestRunEveryMultiTenantExample 2024-04-03T16:19:39Z logger.go:66: │   with module.vpes.ibm_is_subnet_reserved_ip.ip["vpe-allmt-zlrco2-vpc-instance-subnet-b-hs-crypto-tke-gateway-2-ip"],
TestRunEveryMultiTenantExample 2024-04-03T16:19:39Z logger.go:66: │   on ../../main.tf line 74, in resource "ibm_is_subnet_reserved_ip" "ip":
TestRunEveryMultiTenantExample 2024-04-03T16:19:39Z logger.go:66: │   74:   name   = each.value.name
TestRunEveryMultiTenantExample 2024-04-03T16:19:39Z logger.go:66: │ 
TestRunEveryMultiTenantExample 2024-04-03T16:19:39Z logger.go:66: ╵
TestRunEveryMultiTenantExample 2024-04-03T16:19:39Z logger.go:66: ╷
TestRunEveryMultiTenantExample 2024-04-03T16:19:39Z logger.go:66: │ Error: "name" ("vpe-allmt-zlrco2-vpe-allmt-zlrco2-vpc-instance-vpc-hs-crypto-ep11-az3-3") must contain from 1 to 63 characters 
TestRunEveryMultiTenantExample 2024-04-03T16:19:39Z logger.go:66: │ 
TestRunEveryMultiTenantExample 2024-04-03T16:19:39Z logger.go:66: │   with module.vpes.ibm_is_subnet_reserved_ip.ip["vpe-allmt-zlrco2-vpc-instance-subnet-c-hs-crypto-ep11-az3-gateway-3-ip"],
TestRunEveryMultiTenantExample 2024-04-03T16:19:39Z logger.go:66: │   on ../../main.tf line 74, in resource "ibm_is_subnet_reserved_ip" "ip":
TestRunEveryMultiTenantExample 2024-04-03T16:19:39Z logger.go:66: │   74:   name   = each.value.name
TestRunEveryMultiTenantExample 2024-04-03T16:19:39Z logger.go:66: │ 
TestRunEveryMultiTenantExample 2024-04-03T16:19:39Z logger.go:66: ╵
TestRunEveryMultiTenantExample 2024-04-03T16:19:39Z logger.go:66: ╷
TestRunEveryMultiTenantExample 2024-04-03T16:19:39Z logger.go:66: │ Error: "name" ("vpe-allmt-zlrco2-vpe-allmt-zlrco2-vpc-instance-vpc-hs-crypto-tke-1") must contain from 1 to 63 characters 
TestRunEveryMultiTenantExample 2024-04-03T16:19:39Z logger.go:66: │ 
TestRunEveryMultiTenantExample 2024-04-03T16:19:39Z logger.go:66: │   with module.vpes.ibm_is_subnet_reserved_ip.ip["vpe-allmt-zlrco2-vpc-instance-subnet-a-hs-crypto-tke-gateway-1-ip"],
TestRunEveryMultiTenantExample 2024-04-03T16:19:39Z logger.go:66: │   on ../../main.tf line 74, in resource "ibm_is_subnet_reserved_ip" "ip":
TestRunEveryMultiTenantExample 2024-04-03T16:19:39Z logger.go:66: │   74:   name   = each.value.name
TestRunEveryMultiTenantExample 2024-04-03T16:19:39Z logger.go:66: │ 
TestRunEveryMultiTenantExample 2024-04-03T16:19:39Z logger.go:66: ╵
TestRunEveryMultiTenantExample 2024-04-03T16:19:39Z logger.go:66: ╷
TestRunEveryMultiTenantExample 2024-04-03T16:19:39Z logger.go:66: │ Error: "name" ("vpe-allmt-zlrco2-vpe-allmt-zlrco2-vpc-instance-vpc-hs-crypto-ep11-az3-2") must contain from 1 to 63 characters 
TestRunEveryMultiTenantExample 2024-04-03T16:19:39Z logger.go:66: │ 
TestRunEveryMultiTenantExample 2024-04-03T16:19:39Z logger.go:66: │   with module.vpes.ibm_is_subnet_reserved_ip.ip["vpe-allmt-zlrco2-vpc-instance-subnet-b-hs-crypto-ep11-az3-gateway-2-ip"],
TestRunEveryMultiTenantExample 2024-04-03T16:19:39Z logger.go:66: │   on ../../main.tf line 74, in resource "ibm_is_subnet_reserved_ip" "ip":
TestRunEveryMultiTenantExample 2024-04-03T16:19:39Z logger.go:66: │   74:   name   = each.value.name
TestRunEveryMultiTenantExample 2024-04-03T16:19:39Z logger.go:66: │ 
TestRunEveryMultiTenantExample 2024-04-03T16:19:39Z logger.go:66: ╵
```

**Comment by marifse** [comment]
@ocofaigh can you please try rereun the pipeline now.  Thanks

**Comment by ocofaigh** [comment]
Still failing with this (cc @toddgiguere ):
```
TestRunDefaultExample 2024-04-05T13:47:12Z retry.go:99: Returning due to fatal error: FatalError{Underlying: error while running command: exit status 1; ╷
│ Error: "name" ("vpe-default-e8zlkr-vpe-default-e8zlkr-vpc-cloud-object-storage-3") must contain from 1 to 63 characters 
│ 
│   with module.vpes.ibm_is_subnet_reserved_ip.ip["vpe-default-e8zlkr-vpc-instance-subnet-c-cloud-object-storage-gateway-3-ip"],
│   on ../../main.tf line 74, in resource "ibm_is_subnet_reserved_ip" "ip":
│   74:   name   = each.value.name
│ 
╵
╷
│ Error: "name" ("vpe-default-e8zlkr-vpe-default-e8zlkr-vpc-cloud-object-storage-1") must contain from 1 to 63 characters 
│ 
│   with module.vpes.ibm_is_subnet_reserved_ip.ip["vpe-default-e8zlkr-vpc-instance-subnet-a-cloud-object-storage-gateway-1-ip"],
│   on ../../main.tf line 74, in resource "ibm_is_subnet_reserved_ip" "ip":
│   74:   name   = each.value.name
│ 
╵
╷
│ Error: "name" ("vpe-default-e8zlkr-vpe-default-e8zlkr-vpc-cloud-object-storage-2") must contain from 1 to 63 characters 
│ 
│   with module.vpes.ibm_is_subnet_reserved_ip.ip["vpe-default-e8zlkr-vpc-instance-subnet-b-cloud-object-storage-gateway-2-ip"],
│   on ../../main.tf line 74, in resource "ibm_is_subnet_reserved_ip" "ip":
│   74:   name   = each.value.name
│ 
╵}
```

**Comment by marifse** [comment]
@ocofaigh can you please try rerunning the pipeline

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**